src/main/kotlin/org/bundleproject/plugins/Routing.kt

```diff
@@ -1,15 +1,99 @@
 package org.bundleproject.plugins
 
 import io.ktor.routing.*
-import io.ktor.http.*
 import io.ktor.application.*
+import io.ktor.http.*
 import io.ktor.response.*
-import io.ktor.request.*
+import org.bundleproject.json.ModData
+import org.bundleproject.json.responses.ErrorResponse
+import org.bundleproject.json.responses.ModResponse
+import org.bundleproject.json.responses.ModResponseData
+import org.bundleproject.utils.ModNotFoundException
+import org.bundleproject.utils.fetchAssets
+import org.bundleproject.utils.resolveUrl
 
 fun Application.configureRouting() {
     routing {
         get("/") {
-            call.respondText("Hello World!")
+            TODO("Send docs here")
+        }
+        get("/v1/mods/{id}/{platform}/{minecraftVersion}/{version}") {
+            val assets = fetchAssets()
+            val id = call.parameters["id"]!!
+            val platform = call.parameters["platform"]!!
+            val minecraftVersion = call.parameters["minecraftVersion"]!!
+            val version = call.parameters["version"]!!
+            val mod = assets.mods[id]
+            val modData = mod
+                ?.platforms
+                ?.get(platform)
+                ?.get(minecraftVersion)
+                ?.get(version)
+                ?: return@get call.respond(
+                    HttpStatusCode.NotFound,
+                    ErrorResponse(
+                        error = "Invalid mod"
+                    )
+                )
+            try {
+                call.respond(
+                    HttpStatusCode.OK,
+                    ModResponse(
+                        data = ModResponseData(
+                            url = resolveUrl(ModData(
+                                version = version,
+                                source = modData.source,
+                                ref = modData.ref,
+                                name = id,
+                                id = modData.id
+                            )),
+                            metadata = mod.metadata
+                        )
+                    )
+                )
+            } catch (ignored: ModNotFoundException) {
```

Comment:
Instead of using a try-catch would utilise https://ktor.io/docs/status-pages.html

Reply:
I think the current method is better, it returns json data along with the 500 code, but if you think that is unnecessary ill remove it

Reply:
It's not that it returns it is bad but it's rather that it would be cleaner than repetitive code